src/Orleans.Runtime/Catalog/Catalog.cs

```diff
@@ -276,33 +281,51 @@ public void UnregisterSystemTarget(ISystemTarget target)
             } // End lock
 
             if (result is null)
+            {
+                rehydrationContext?.Dispose();
+                return UnableToCreateActivation(this, grainId);
+            }
+
+            // Rehydration occurs before activation.
+            if (rehydrationContext is not null)
+            {
+                result.Rehydrate(rehydrationContext);
```

Comment:
Should we surround this with a try/catch block? If the `Rehydrate` fails, I assume we don't want to keep this activation